src/lib/rest/httpRequestSend.cpp

```diff
@@ -573,7 +576,17 @@ int httpRequestSend
   {
     url = ip;
   }
-  url = protocol + url + ":" + portAsString + (resource.at(0) == '/'? "" : "/") + resource;
+
+  if (apiVersion == V2 && verb == "POST" && resource == "/v2/op/query")
+  {
+    std::string pLimit = std::to_string(providerLimit);
+    std::string pOffset = std::to_string(providerOffset);
+    url = protocol + url + ":" + portAsString + (resource.at(0) == '/'? "" : "/") + resource + "?limit=" + pLimit + "&offset=" + pOffset + "&options=count";
+  }
+  else
+  {
+    url = protocol + url + ":" + portAsString + (resource.at(0) == '/'? "" : "/") + resource;
+  }
```

Comment:
Related with previous comments (apiVersion not needed and >=0 to mean you want to include some of the new URL query parameters), I'd suggest to do something like this:

```suggestion
 url = protocol + url + ":" + portAsString + (resource.at(0) == '/'? "" : "/") + resource;

if (providerLimit >= 0) && (providerOffset >=0)
{
  std::string pLimit = std::to_string(providerLimit);
  std::string pOffset = std::to_string(providerOffset);
  url += "?limit=" + pLimit + "&offset=" + pOffset + "&options=count";
}
```

Reply:
Fixed in 1e47b07